frontend/src/components/templates/Admin/UserList/index.js

```diff
@@ -74,26 +78,31 @@ const UserList = () => {
     );
   };
 
-  useEffect(() => {
-    setSkipReset(false);
-  }, [users]);
+  const fetchUsers = useCallback(async () => {
+    setIsLoading(true);
+    const {
+      users,
+      meta: { total_pages },
+    } = await getUsers({
+      usersWithRolesOnly: !showAllUsers,
+      page: pagination.pageIndex,
+      filter: filtered,
+    });
+    setIsLoading(false);
+    setUsers(users);
+    setTotalPages(total_pages);
+  }, [showAllUsers, pagination, filtered]);
 
   useEffect(() => {
-    const onFetchData = async () => {
-      setIsLoading(true);
-      const { users } = await getUsers({ usersWithRolesOnly: !showAllUsers });
-      setUsers(users);
-      setIsLoading(false);
-    };
-    onFetchData();
-  }, [showAllUsers]);
+    const debouncedFetchData = debounce(() => {
+      fetchUsers();
+    }, 1500);
```

Comment:
1500 is far to long a duration. You should use 100ms here like in frontend/src/components/templates/InstructorEnrollmentList.tsx